pkg/controller/podautoscaler/horizontal.go

```diff
@@ -1006,27 +1006,27 @@ func calculateScaleUpLimitWithScalingRules(currentReplicas int32, scaleEvents []
 		replicasAddedInCurrentPeriod := getReplicasChangePerPeriod(policy.PeriodSeconds, scaleEvents)
 		periodStartReplicas := currentReplicas - replicasAddedInCurrentPeriod
 		if policy.Type == autoscalingv2.PodsScalingPolicy {
-			proposed = int32(periodStartReplicas + policy.Value)
+			proposed = periodStartReplicas + policy.Value
 		} else if policy.Type == autoscalingv2.PercentScalingPolicy {
 			// the proposal has to be rounded up because the proposed change might not increase the replica count causing the target to never scale up
 			proposed = int32(math.Ceil(float64(periodStartReplicas) * (1 + float64(policy.Value)/100)))
 		}
-		result = selectPolicyFn(result, proposed)
+		steps = append(steps,proposed)
 	}
-	return result
+	return selectPolicyFn(steps)
 }
 
 // calculateScaleDownLimitWithBehavior returns the maximum number of pods that could be deleted for the given HPAScalingRules
 func calculateScaleDownLimitWithBehaviors(currentReplicas int32, scaleEvents []timestampedScaleEvent, scalingRules *autoscalingv2.HPAScalingRules) int32 {
-	var result int32 = math.MaxInt32
+	var steps []int32
 	var proposed int32
-	var selectPolicyFn func(int32, int32) int32
+	var selectPolicyFn func([]int32) int32
 	if *scalingRules.SelectPolicy == autoscalingv2.DisabledPolicySelect {
 		return currentReplicas // Scaling is disabled
 	} else if *scalingRules.SelectPolicy == autoscalingv2.MinPolicySelect {
-		selectPolicyFn = max // For scaling down, the lowest change ('min' policy) produces a maximum value
+		selectPolicyFn = min // For scaling down, the lowest change ('min' policy) produces a minimum value
```

Comment:
`Minimum` in the policy means minimum change in the number of replicas. So the code was doing the right thing previously.

Reply:
you mean this? :
    *scalingRules.SelectPolicy == autoscalingv2.MinPolicySelect   selectPolicyFn = max
    *scalingRules.SelectPolicy == autoscalingv2.MaxPolicySelect   selectPolicyFn = min

Reply:
Yes, that's right. But the functions `min` and `max` do not directly do `MinPolicySelect` and `MaxPolicySelect`. Here's why. Assume you have 5 replicas and you have 2 policies: one suggests that that the deployment should scale down to 3 replica and the other recommends 4 replicas. Now if you use `MaxPolicySelect` the controller picks the recommendation which has the maximum change, in this case the first recommendation because the change in replicas would be 5-3=2. But to determine that you need the the `min` of all the proposed changes. So the `min` in this case corresponds to `MaxPolicySelect` and similarly `max` corresponds to `MinPolicySelect`. The function names are bad but they do what they are supposed to do.

Reply:
Thank you very much for pointing out the problem. I'll revise it immediately

Reply:
@arjunrn  I had fixed the problem above, do you have any other suggestions？

Reply:
I think that `Min` and `Max` alone have proved to be confusing.  I would recommend changing them to `MinChange` and `MaxChange` with the graduation of v2beta2 to stable: https://docs.google.com/document/d/1QGT9Waxc500ICGX_TGIoxfOWKmWbXYC6TWxfdUC2Njg/edit#heading=h.u4pozdbhwg30

Reply:
@josephburnett  Yeah, I'll revise it